drivers/net/bnxt/bnxt_hwrm.c

```diff
@@ -2473,6 +2497,8 @@ int bnxt_hwrm_func_vf_vnic_query_and_config(struct bnxt *bp, uint16_t vf,
 		rc = bnxt_hwrm_vnic_qcfg(bp, &vnic, bp->pf.first_vf_id + vf);
 		if (rc)
 			break;
+		if (vnic.mru == 4)	// Unallocated
+			continue;
```

Comment:
There were some firmware issues with us doing anything with unallocated VNICs.  This prevents that from happening.